crates/weaver_resolved_schema/src/v2/mod.rs

```diff
@@ -0,0 +1,604 @@
+//! Version 2 of semantic convention schema.
+
+use std::collections::{HashMap, HashSet};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::{
+    group::GroupType,
+    v2::{signal_id::SignalId, span::SpanName, CommonFields},
+};
+
+use crate::v2::{
+    catalog::Catalog,
+    entity::Entity,
+    metric::Metric,
+    registry::Registry,
+    span::{Span, SpanRefinement},
+};
+
+pub mod attribute;
+pub mod catalog;
+pub mod entity;
+pub mod event;
+pub mod metric;
+pub mod registry;
+pub mod span;
+
+/// A Resolved Telemetry Schema.
+/// A Resolved Telemetry Schema is self-contained and doesn't contain any
+/// external references to other schemas or semantic conventions.
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct ResolvedTelemetrySchema {
+    /// Version of the file structure.
+    pub file_format: String,
+    /// Schema URL that this file is published at.
+    pub schema_url: String,
+    /// The ID of the registry that this schema belongs to.
+    pub registry_id: String,
+    /// The registry that this schema belongs to.
+    pub registry: Registry,
+    /// Catalog of unique items that are shared across multiple registries
+    /// and signals.
+    pub catalog: Catalog,
+    // TODO - versions, dependencies and other options.
+}
+
+fn fix_group_id(prefix: &'static str, group_id: &str) -> SignalId {
+    if group_id.starts_with(prefix) {
+        group_id.trim_start_matches(prefix).to_owned().into()
+    } else {
+        group_id.to_owned().into()
+    }
+}
+
+fn fix_span_group_id(group_id: &str) -> SignalId {
+    fix_group_id("span.", group_id)
+}
+
+/// Converts a V1 registry + catalog to V2.
+pub fn convert_v1_to_v2(
+    c: crate::catalog::Catalog,
+    r: crate::registry::Registry,
+) -> Result<(Catalog, Registry), crate::error::Error> {
+    // When pulling attributes, as we collapse things, we need to filter
+    // to just unique.
```

Comment:
I don't understand this comment unless we consider that equality between two attributes doesn't follow the same rules between v1 and v2. Is that the case? And if so, what's the rationale?

Reply:
The removal of `requirement_level` actually leads to a decent "compression" or removal of duplicate attribute definitions that diverge only in requirement level.

In V2, this is critical, as we want *attribute key* to be a namespace for attributes.  In V1 - a pure 'attribute key' is actually ambiguous in the catalog.